packages/widgets/src/modals/dynamic-edit-modal.tsx

```diff
@@ -0,0 +1,66 @@
+"use client";
+
+import { Button, Group, Stack, TextInput } from "@mantine/core";
+import { z } from "zod";
+
+import { createModal } from "@homarr/modals";
+import { useI18n } from "@homarr/translation/client";
+import { zodErrorMap } from "@homarr/validation/form";
+
+import { FormProvider, useForm } from "../_inputs/dynamic-form";
+
+export interface DynamicEditModalState {
+  options: object;
+}
+
+interface ModalProps {
+  kind: ["dynamic"];
+  value: DynamicEditModalState;
+  onSuccessfulEdit: (value: DynamicEditModalState) => void;
```

Comment:
Also change to  `z.infer<typeof dynamicSectionOptionsSchema>` for the type of value